### PR TITLE
[Microsoft API] generate_login_url - support prompt=consent

### PR DESCRIPTION
**Pull request by michal-dagan**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-9390)

## Description
Added prompt=consent triggers the OAuth consent dialog after the user signs in, asking the user to grant permissions to the app.
https://learn.microsoft.com/en-us/entra/identity-platform/v2-oauth2-auth-code-flow
## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by DeanArbel — APPROVED** [review]
Good job

File: Packs/MicrosoftGraphAPI/ReleaseNotes/1_1_39.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Microsoft Graph API
+
+- Added support for OAuth consent dialog after the user signs in using the ***msgraph-api-generate-login-url*** command.
```
Comment (michal-dagan) [inline Packs/MicrosoftGraphAPI/ReleaseNotes/1_1_39.md:6]:
```suggestion
Updated the **MicrosoftApiModule** to better handle the Authorization Code flow in the supported integrations.
```

File: Packs/AzureKubernetesServices/ReleaseNotes/1_1_22.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Azure Kubernetes Services
+
+- Added support for OAuth consent dialog after the user signs in using the ***azure-ks-generate-login-url*** command.
```
Comment (ShirleyDenkberg) [inline Packs/AzureKubernetesServices/ReleaseNotes/1_1_22.md:6]:
```suggestion
Added support for the OAuth consent dialog after the user signs in using the ***azure-ks-generate-login-url*** command.
```

File: Packs/AzureLogAnalytics/ReleaseNotes/1_1_24.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Azure Log Analytics
+
+- Added support for OAuth consent dialog after the user signs in using the ***azure-log-analytics-generate-login-url*** command.
```
Comment (ShirleyDenkberg) [inline Packs/AzureLogAnalytics/ReleaseNotes/1_1_24.md:6]:
```suggestion
Added support for the OAuth consent dialog after the user signs in using the ***azure-log-analytics-generate-login-url*** command.
```

File: Packs/AzureNetworkSecurityGroups/ReleaseNotes/1_2_23.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Azure Network Security Groups
+
+- Added support for OAuth consent dialog after the user signs in using the ***azure-nsg-generate-login-url*** command.
```
Comment (ShirleyDenkberg) [inline Packs/AzureNetworkSecurityGroups/ReleaseNotes/1_2_23.md:6]:
```suggestion
Added support for the OAuth consent dialog after the user signs in using the ***azure-nsg-generate-login-url*** command.
```

File: Packs/AzureSQLManagement/ReleaseNotes/1_1_39.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Azure SQL Management
+
+- Added support for OAuth consent dialog after the user signs in using the ***azure-sql-generate-login-url*** command.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSQLManagement/ReleaseNotes/1_1_39.md:6]:
```suggestion
Added support for the OAuth consent dialog after the user signs in using the ***azure-sql-generate-login-url*** command.
```

File: Packs/AzureStorage/ReleaseNotes/1_2_22.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Azure Storage Management
+
+- Added support for OAuth consent dialog after the user signs in using the ***azure-storage-generate-login-url*** command.
```
Comment (ShirleyDenkberg) [inline Packs/AzureStorage/ReleaseNotes/1_2_22.md:6]:
```suggestion
Added support for the OAuth consent dialog after the user signs in using the ***azure-storage-generate-login-url*** command.
```

File: Packs/AzureWAF/ReleaseNotes/1_1_20.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Azure Web Application Firewall
+
+- Added support for OAuth consent dialog after the user signs in using the ***azure-waf-generate-login-url*** command.
```
Comment (ShirleyDenkberg) [inline Packs/AzureWAF/ReleaseNotes/1_1_20.md:6]:
```suggestion
Added support for the OAuth consent dialog after the user signs in using the ***azure-waf-generate-login-url*** command.
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_16_24.md [hunk 8]
```diff
@@ -0,0 +1,10 @@
+
+#### Integrations
+
+##### Microsoft Defender for Endpoint
+
+- Added support for OAuth consent dialog after the user signs in using the ***microsoft-atp-generate-login-url*** command.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_16_24.md:6]:
```suggestion
Added support for the OAuth consent dialog after the user signs in using the ***microsoft-atp-generate-login-url*** command.
```

File: Packs/MicrosoftExchangeOnline/ReleaseNotes/1_2_37.md [hunk 9]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### EWS O365
+
+- Added support for OAuth consent dialog after the user signs in using the integration command prefix-generate-login-url command.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftExchangeOnline/ReleaseNotes/1_2_37.md:6]:
```suggestion
Added support for the OAuth consent dialog after the user signs in using the integration command ***prefix-generate-login-url*** command.
```

File: Packs/MicrosoftGraphFiles/ReleaseNotes/1_1_22.md [hunk 10]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### O365 File Management (Onedrive/Sharepoint/Teams)
+
+- Added support for OAuth consent dialog after the user signs in using the ***msgraph-files-generate-login-url*** command.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphFiles/ReleaseNotes/1_1_22.md:6]:
```suggestion
Added support for the OAuth consent dialog after the user signs in using the ***msgraph-files-generate-login-url*** command.
```

File: Packs/MicrosoftGraphGroups/ReleaseNotes/1_1_40.md [hunk 11]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Azure Active Directory Groups
+
+- Added support for OAuth consent dialog after the user signs in using the ***msgraph-groups-generate-login-url*** command.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphGroups/ReleaseNotes/1_1_40.md:6]:
```suggestion
Added support for the OAuth consent dialog after the user signs in using the ***msgraph-groups-generate-login-url*** command.
```

File: Packs/MicrosoftGraphMail/ReleaseNotes/1_6_2.md [hunk 12]
```diff
@@ -0,0 +1,10 @@
+
+#### Integrations
+
+##### O365 Outlook Mail (Using Graph API)
+
+Updated the **MicrosoftApiModule** to better handle the Authorization Code flow in the supported integrations.
+
+##### Microsoft Graph Mail Single User
+
+- Added support for OAuth consent dialog after the user signs in using the ***msgraph-mail-generate-login-url*** command.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphMail/ReleaseNotes/1_6_2.md:10]:
```suggestion
Added support for the OAuth consent dialog after the user signs in using the ***msgraph-mail-generate-login-url*** command.
```

File: Packs/MicrosoftGraphSearch/ReleaseNotes/1_0_8.md [hunk 13]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Microsoft Graph Search
+
+- Added support for OAuth consent dialog after the user signs in using the ***msgraph-search-generate-login-url*** command.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSearch/ReleaseNotes/1_0_8.md:6]:
```suggestion
Added support for the OAuth consent dialog after the user signs in using the ***msgraph-search-generate-login-url*** command.
```

File: Packs/MicrosoftGraphSecurity/ReleaseNotes/2_2_7.md [hunk 14]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Microsoft Graph Security
+
+- Added support for OAuth consent dialog after the user signs in using the ***msg-generate-login-url*** command.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftGraphSecurity/ReleaseNotes/2_2_7.md:6]:
```suggestion
Added support for the OAuth consent dialog after the user signs in using the ***msg-generate-login-url*** command.
```

**Comment by ShirleyDenkberg** [comment]
@DeanArbel Doc review completed.

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml [hunk 15]
```diff
@@ -712,7 +712,7 @@ script:
   - description: Generate the login url used for Authorization code flow.
     name: microsoft-teams-generate-login-url
     arguments: []
-  dockerimage: demisto/teams:1.0.0.86933
+  dockerimage: demisto/teams:1.0.0.87469
```
Comment (michal-dagan) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml:715]:
```suggestion
  dockerimage: demisto/teams:1.0.0.87635
```

File: Packs/MicrosoftTeams/ReleaseNotes/1_4_51.md [hunk 16]
```diff
@@ -0,0 +1,11 @@
+
+#### Integrations
+
+##### Microsoft Teams
+
+- Added support for the OAuth consent dialog after the user signs in using the ***microsoft-teams-generate-login-url*** command.
+- Updated the Docker image to: *demisto/teams:1.0.0.87469*.
```
Comment (michal-dagan) [inline Packs/MicrosoftTeams/ReleaseNotes/1_4_51.md:7]:
```suggestion
- Updated the Docker image to: *demisto/teams:1.0.0.87635*.
```

File: Packs/MicrosoftTeams/ReleaseNotes/1_4_51.md [hunk 17]
```diff
@@ -0,0 +1,11 @@
+
+#### Integrations
+
+##### Microsoft Teams
+
+- Added support for the OAuth consent dialog after the user signs in using the ***microsoft-teams-generate-login-url*** command.
+- Updated the Docker image to: *demisto/teams:1.0.0.87635*.
```
Comment (michal-dagan) [inline Packs/MicrosoftTeams/ReleaseNotes/1_4_51.md:7]:
```suggestion
- Updated the Docker image to: *demisto/teams:1.0.0.87877*.
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml [hunk 18]
```diff
@@ -712,7 +712,7 @@ script:
   - description: Generate the login url used for Authorization code flow.
     name: microsoft-teams-generate-login-url
     arguments: []
-  dockerimage: demisto/teams:1.0.0.86933
+  dockerimage: demisto/teams:1.0.0.87635
```
Comment (michal-dagan) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml:715]:
```suggestion
  dockerimage: demisto/teams:1.0.0.87877
```

**Comment by michal-dagan** [comment]
@DeanArbel 
Force merge is needed due to the following issue: [CIAC-9788](https://jira-dc.paloaltonetworks.com/browse/CIAC-9788).
in [xsoar_ng_server_ga](https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/jobs/4833054)

File: Tests/conf.json [hunk 19]
```diff
@@ -5866,7 +5866,8 @@
         "MSG-Threat-Assessment-test": "API limitation",
         "BambenekConsultingFeed_Test": "Issue CRTX-99480",
         "AWS SNS Listener - Test": "Cant validate mock msg against AWS-SNS in TBP",
-        "MSG-ediscovery-tpb": "Issue CIAC-9763"
+        "MSG-ediscovery-tpb": "Issue CIAC-9763",
```
Comment (michal-dagan) [inline Tests/conf.json:5869]:
```suggestion
        "MSG-ediscovery-tpb": "Issue CIAC-9763"
```

File: Tests/conf.json [hunk 20]
```diff
@@ -5866,7 +5866,8 @@
...
+        "Azure Kubernetes Services - Test": "CRTX-106591"
```
Comment (michal-dagan) [inline Tests/conf.json:5870]:
```suggestion
```

**Pull request merged**